### PR TITLE
Jest tests for routeTree code

### PR DESCRIPTION
**Pull request by chromakode**
:eyeglasses: @keybase/react-hackers 

:warning: Ran into two hard flow issues while writing tests; these aren't fixed yet and I'd like to ask @MarcoPolo in person about them tomorrow.

This adds full test coverage for the core data structure code in `route-tree/index.js` and the nontrivial reducer actions.

image

It also adds some gross config hacks:

Getting jest working with our file structure was not pretty. :sob: First I had to hack around the typical issues with not using a `.babelrc` and not having babel modules relative to a `.babelrc`. I created a new transformer that configures babel with resolved module paths based on configuration shared with webpack.

Also, the coverage testing library jest uses (istanbul) doesn't instrument files above the cwd. My first approach was to hack around this. I found a really gross ~tweak to our `package.json`~ monkeypatch in the babel transformer that will trick babel-istanbul into thinking the cwd is `client/`, not `desktop/`, which makes coverage work. This felt really dodgy, so I came up with a second approach: make `shared/` the jest top level directory. However, this comes with its own issue: the damn istanbul instrumenter plugin looks for a `package.json` file in a parent directory, and crashes when it can't find one. We could propose a patch to fix this behavior, but for the time being the best mitigation is to include a package.json in shared (this is also maybe a step in the right direction; making shared its own bona fide npm package might be a solution to our fs layout tooling issues). Initially I just made a stub `package.json` with name and description, but *then* found out eslint complains if the `package.json` doesn't contain all the referenced deps, so I put the real deps in there.

Long story short, this required some gross configuration hacks because of our nonstandard fs layout. I've tried to minimize them, but they're still gross. :disappointed: I believe that coverage testing is really useful to the test development process, so it's worth having these hacks for now to have full coverage functionality.

### CODE REVIEW

File: shared/package.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
```
Comment (chrisnojima) [inline shared/package.json]: i'm worried about this and it being out of date. if we mark the values as "0.0.0" does that still make eslint happy?

**Comment by keybase-ci-visdiff** [comment]
The commits c70ceccceb9a...a0e3a1d364e4 introduce visual changes on linux. <details><summary>:mag_right: 4 new, 3 removed, 3 changed</summary>

 * Added: **Settings_Container-Normal_Good_Banner**  
   Settings_Container-Normal_Good_Banner rendered
 * Added: **Settings_Container-Normal_Bad_Banner**  
   Settings_Container-Normal_Bad_Banner rendered
 * Added: **Settings_Container-Normal**  
   Settings_Container-Normal rendered
 * Added: **Profile-Your_Profile_Following_Tab**  
   Profile-Your_Profile_Following_Tab rendered
 * Changed: **Folders_TLF-Normal_Public** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479851231754-c70ceccceb9a-a0e3a1d364e4-linux/Folders_TLF-Normal_Public-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479851231754-c70ceccceb9a-a0e3a1d364e4-linux/Folders_TLF-Normal_Public-a0e3a1d364e4.png)  
   Folders_TLF-Normal_Public comparison
 * Changed: **App_Tab_Bar-Search_Active** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479851231754-c70ceccceb9a-a0e3a1d364e4-linux/App_Tab_Bar-Search_Active-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479851231754-c70ceccceb9a-a0e3a1d364e4-linux/App_Tab_Bar-Search_Active-a0e3a1d364e4.png)  
   App_Tab_Bar-Search_Active comparison
 * Changed: **App_Tab_Bar-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479851231754-c70ceccceb9a-a0e3a1d364e4-linux/App_Tab_Bar-Normal-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479851231754-c70ceccceb9a-a0e3a1d364e4-linux/App_Tab_Bar-Normal-a0e3a1d364e4.png)  
   App_Tab_Bar-Normal comparison
 * Removed: **Settings_Nav-Normal_Good_Banner** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479851231754-c70ceccceb9a-a0e3a1d364e4-linux/Settings_Nav-Normal_Good_Banner-c70ceccceb9a.png)
 * Removed: **Settings_Nav-Normal_Bad_Banner** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479851231754-c70ceccceb9a-a0e3a1d364e4-linux/Settings_Nav-Normal_Bad_Banner-c70ceccceb9a.png)
 * Removed: **Settings_Nav-Normal** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479851231754-c70ceccceb9a-a0e3a1d364e4-linux/Settings_Nav-Normal-c70ceccceb9a.png)
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits c70ceccceb9a...a0e3a1d364e4 introduce visual changes on linux. <details><summary>:mag_right: 4 new, 3 removed, 3 changed</summary>

 * Added: **Settings_Container-Normal_Good_Banner**  
   Settings_Container-Normal_Good_Banner rendered
 * Added: **Settings_Container-Normal_Bad_Banner**  
   Settings_Container-Normal_Bad_Banner rendered
 * Added: **Settings_Container-Normal**  
   Settings_Container-Normal rendered
 * Added: **Profile-Your_Profile_Following_Tab**  
   Profile-Your_Profile_Following_Tab rendered
 * Changed: **Folders_TLF-Normal_Public** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853014425-c70ceccceb9a-a0e3a1d364e4-linux/Folders_TLF-Normal_Public-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853014425-c70ceccceb9a-a0e3a1d364e4-linux/Folders_TLF-Normal_Public-a0e3a1d364e4.png)  
   Folders_TLF-Normal_Public comparison
 * Changed: **App_Tab_Bar-Search_Active** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853014425-c70ceccceb9a-a0e3a1d364e4-linux/App_Tab_Bar-Search_Active-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853014425-c70ceccceb9a-a0e3a1d364e4-linux/App_Tab_Bar-Search_Active-a0e3a1d364e4.png)  
   App_Tab_Bar-Search_Active comparison
 * Changed: **App_Tab_Bar-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853014425-c70ceccceb9a-a0e3a1d364e4-linux/App_Tab_Bar-Normal-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853014425-c70ceccceb9a-a0e3a1d364e4-linux/App_Tab_Bar-Normal-a0e3a1d364e4.png)  
   App_Tab_Bar-Normal comparison
 * Removed: **Settings_Nav-Normal_Good_Banner** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853014425-c70ceccceb9a-a0e3a1d364e4-linux/Settings_Nav-Normal_Good_Banner-c70ceccceb9a.png)
 * Removed: **Settings_Nav-Normal_Bad_Banner** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853014425-c70ceccceb9a-a0e3a1d364e4-linux/Settings_Nav-Normal_Bad_Banner-c70ceccceb9a.png)
 * Removed: **Settings_Nav-Normal** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853014425-c70ceccceb9a-a0e3a1d364e4-linux/Settings_Nav-Normal-c70ceccceb9a.png)
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits c70ceccceb9a...0ab5afe3fc9f introduce visual changes on linux. <details><summary>:mag_right: 4 new, 3 removed, 4 changed</summary>

 * Added: **Settings_Container-Normal_Good_Banner**  
   Settings_Container-Normal_Good_Banner rendered
 * Added: **Settings_Container-Normal_Bad_Banner**  
   Settings_Container-Normal_Bad_Banner rendered
 * Added: **Settings_Container-Normal**  
   Settings_Container-Normal rendered
 * Added: **Profile-Your_Profile_Following_Tab**  
   Profile-Your_Profile_Following_Tab rendered
 * Changed: **Paper_Key-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853123609-c70ceccceb9a-0ab5afe3fc9f-linux/Paper_Key-Normal-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853123609-c70ceccceb9a-0ab5afe3fc9f-linux/Paper_Key-Normal-0ab5afe3fc9f.png)  
   Paper_Key-Normal comparison
 * Changed: **Folders_TLF-Normal_Public** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853123609-c70ceccceb9a-0ab5afe3fc9f-linux/Folders_TLF-Normal_Public-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853123609-c70ceccceb9a-0ab5afe3fc9f-linux/Folders_TLF-Normal_Public-0ab5afe3fc9f.png)  
   Folders_TLF-Normal_Public comparison
 * Changed: **App_Tab_Bar-Search_Active** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853123609-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Search_Active-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853123609-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Search_Active-0ab5afe3fc9f.png)  
   App_Tab_Bar-Search_Active comparison
 * Changed: **App_Tab_Bar-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853123609-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Normal-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853123609-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Normal-0ab5afe3fc9f.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853123609-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Normal.png)
 * Removed: **Settings_Nav-Normal_Good_Banner** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853123609-c70ceccceb9a-0ab5afe3fc9f-linux/Settings_Nav-Normal_Good_Banner-c70ceccceb9a.png)
 * Removed: **Settings_Nav-Normal_Bad_Banner** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853123609-c70ceccceb9a-0ab5afe3fc9f-linux/Settings_Nav-Normal_Bad_Banner-c70ceccceb9a.png)
 * Removed: **Settings_Nav-Normal** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479853123609-c70ceccceb9a-0ab5afe3fc9f-linux/Settings_Nav-Normal-c70ceccceb9a.png)
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits c70ceccceb9a...0ab5afe3fc9f introduce visual changes on linux. <details><summary>:mag_right: 4 new, 3 removed, 3 changed</summary>

 * Added: **Settings_Container-Normal_Good_Banner**  
   Settings_Container-Normal_Good_Banner rendered
 * Added: **Settings_Container-Normal_Bad_Banner**  
   Settings_Container-Normal_Bad_Banner rendered
 * Added: **Settings_Container-Normal**  
   Settings_Container-Normal rendered
 * Added: **Profile-Your_Profile_Following_Tab**  
   Profile-Your_Profile_Following_Tab rendered
 * Changed: **Folders_TLF-Normal_Public** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479854346928-c70ceccceb9a-0ab5afe3fc9f-linux/Folders_TLF-Normal_Public-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479854346928-c70ceccceb9a-0ab5afe3fc9f-linux/Folders_TLF-Normal_Public-0ab5afe3fc9f.png)  
   Folders_TLF-Normal_Public comparison
 * Changed: **App_Tab_Bar-Search_Active** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479854346928-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Search_Active-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479854346928-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Search_Active-0ab5afe3fc9f.png)  
   App_Tab_Bar-Search_Active comparison
 * Changed: **App_Tab_Bar-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479854346928-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Normal-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479854346928-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Normal-0ab5afe3fc9f.png)  
   App_Tab_Bar-Normal comparison
 * Removed: **Settings_Nav-Normal_Good_Banner** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479854346928-c70ceccceb9a-0ab5afe3fc9f-linux/Settings_Nav-Normal_Good_Banner-c70ceccceb9a.png)
 * Removed: **Settings_Nav-Normal_Bad_Banner** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479854346928-c70ceccceb9a-0ab5afe3fc9f-linux/Settings_Nav-Normal_Bad_Banner-c70ceccceb9a.png)
 * Removed: **Settings_Nav-Normal** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479854346928-c70ceccceb9a-0ab5afe3fc9f-linux/Settings_Nav-Normal-c70ceccceb9a.png)
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits c70ceccceb9a...0ab5afe3fc9f introduce visual changes on linux. <details><summary>:mag_right: 4 new, 3 removed, 3 changed</summary>

 * Added: **Settings_Container-Normal_Good_Banner**  
   Settings_Container-Normal_Good_Banner rendered
 * Added: **Settings_Container-Normal_Bad_Banner**  
   Settings_Container-Normal_Bad_Banner rendered
 * Added: **Settings_Container-Normal**  
   Settings_Container-Normal rendered
 * Added: **Profile-Your_Profile_Following_Tab**  
   Profile-Your_Profile_Following_Tab rendered
 * Changed: **Folders_TLF-Normal_Public** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479855009167-c70ceccceb9a-0ab5afe3fc9f-linux/Folders_TLF-Normal_Public-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479855009167-c70ceccceb9a-0ab5afe3fc9f-linux/Folders_TLF-Normal_Public-0ab5afe3fc9f.png)  
   Folders_TLF-Normal_Public comparison
 * Changed: **App_Tab_Bar-Search_Active** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479855009167-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Search_Active-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479855009167-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Search_Active-0ab5afe3fc9f.png)  
   App_Tab_Bar-Search_Active comparison
 * Changed: **App_Tab_Bar-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479855009167-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Normal-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479855009167-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Normal-0ab5afe3fc9f.png)  
   App_Tab_Bar-Normal comparison
 * Removed: **Settings_Nav-Normal_Good_Banner** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479855009167-c70ceccceb9a-0ab5afe3fc9f-linux/Settings_Nav-Normal_Good_Banner-c70ceccceb9a.png)
 * Removed: **Settings_Nav-Normal_Bad_Banner** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479855009167-c70ceccceb9a-0ab5afe3fc9f-linux/Settings_Nav-Normal_Bad_Banner-c70ceccceb9a.png)
 * Removed: **Settings_Nav-Normal** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479855009167-c70ceccceb9a-0ab5afe3fc9f-linux/Settings_Nav-Normal-c70ceccceb9a.png)
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits c70ceccceb9a...0ab5afe3fc9f introduce visual changes on linux. <details><summary>:mag_right: 4 new, 3 removed, 3 changed</summary>

 * Added: **Settings_Container-Normal_Good_Banner**  
   Settings_Container-Normal_Good_Banner rendered
 * Added: **Settings_Container-Normal_Bad_Banner**  
   Settings_Container-Normal_Bad_Banner rendered
 * Added: **Settings_Container-Normal**  
   Settings_Container-Normal rendered
 * Added: **Profile-Your_Profile_Following_Tab**  
   Profile-Your_Profile_Following_Tab rendered
 * Changed: **Folders_TLF-Normal_Public** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479859754151-c70ceccceb9a-0ab5afe3fc9f-linux/Folders_TLF-Normal_Public-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479859754151-c70ceccceb9a-0ab5afe3fc9f-linux/Folders_TLF-Normal_Public-0ab5afe3fc9f.png)  
   Folders_TLF-Normal_Public comparison
 * Changed: **App_Tab_Bar-Search_Active** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479859754151-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Search_Active-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479859754151-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Search_Active-0ab5afe3fc9f.png)  
   App_Tab_Bar-Search_Active comparison
 * Changed: **App_Tab_Bar-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479859754151-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Normal-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479859754151-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Normal-0ab5afe3fc9f.png)  
   App_Tab_Bar-Normal comparison
 * Removed: **Settings_Nav-Normal_Good_Banner** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479859754151-c70ceccceb9a-0ab5afe3fc9f-linux/Settings_Nav-Normal_Good_Banner-c70ceccceb9a.png)
 * Removed: **Settings_Nav-Normal_Bad_Banner** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479859754151-c70ceccceb9a-0ab5afe3fc9f-linux/Settings_Nav-Normal_Bad_Banner-c70ceccceb9a.png)
 * Removed: **Settings_Nav-Normal** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479859754151-c70ceccceb9a-0ab5afe3fc9f-linux/Settings_Nav-Normal-c70ceccceb9a.png)
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits c70ceccceb9a...0ab5afe3fc9f introduce visual changes on linux. <details><summary>:mag_right: 4 new, 3 removed, 3 changed</summary>

 * Added: **Settings_Container-Normal_Good_Banner**  
   Settings_Container-Normal_Good_Banner rendered
 * Added: **Settings_Container-Normal_Bad_Banner**  
   Settings_Container-Normal_Bad_Banner rendered
 * Added: **Settings_Container-Normal**  
   Settings_Container-Normal rendered
 * Added: **Profile-Your_Profile_Following_Tab**  
   Profile-Your_Profile_Following_Tab rendered
 * Changed: **Folders_TLF-Normal_Public** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479859768939-c70ceccceb9a-0ab5afe3fc9f-linux/Folders_TLF-Normal_Public-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479859768939-c70ceccceb9a-0ab5afe3fc9f-linux/Folders_TLF-Normal_Public-0ab5afe3fc9f.png)  
   Folders_TLF-Normal_Public comparison
 * Changed: **App_Tab_Bar-Search_Active** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479859768939-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Search_Active-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479859768939-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Search_Active-0ab5afe3fc9f.png)  
   App_Tab_Bar-Search_Active comparison
 * Changed: **App_Tab_Bar-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479859768939-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Normal-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479859768939-c70ceccceb9a-0ab5afe3fc9f-linux/App_Tab_Bar-Normal-0ab5afe3fc9f.png)  
   App_Tab_Bar-Normal comparison
 * Removed: **Settings_Nav-Normal_Good_Banner** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479859768939-c70ceccceb9a-0ab5afe3fc9f-linux/Settings_Nav-Normal_Good_Banner-c70ceccceb9a.png)
 * Removed: **Settings_Nav-Normal_Bad_Banner** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479859768939-c70ceccceb9a-0ab5afe3fc9f-linux/Settings_Nav-Normal_Bad_Banner-c70ceccceb9a.png)
 * Removed: **Settings_Nav-Normal** [(view original)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479859768939-c70ceccceb9a-0ab5afe3fc9f-linux/Settings_Nav-Normal-c70ceccceb9a.png)
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits c70ceccceb9a...a28a47695855 introduce visual changes on linux. <details><summary>:mag_right: 1 new, 6 changed</summary>

 * Added: **Profile-Your_Profile_Following_Tab**  
   Profile-Your_Profile_Following_Tab rendered
 * Changed: **Settings_Nav-Normal_Good_Banner** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479860518327-c70ceccceb9a-a28a47695855-linux/Settings_Nav-Normal_Good_Banner-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479860518327-c70ceccceb9a-a28a47695855-linux/Settings_Nav-Normal_Good_Banner-a28a47695855.png)  
   Settings_Nav-Normal_Good_Banner comparison
 * Changed: **Settings_Nav-Normal_Bad_Banner** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479860518327-c70ceccceb9a-a28a47695855-linux/Settings_Nav-Normal_Bad_Banner-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479860518327-c70ceccceb9a-a28a47695855-linux/Settings_Nav-Normal_Bad_Banner-a28a47695855.png)  
   Settings_Nav-Normal_Bad_Banner comparison
 * Changed: **Settings_Nav-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479860518327-c70ceccceb9a-a28a47695855-linux/Settings_Nav-Normal-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479860518327-c70ceccceb9a-a28a47695855-linux/Settings_Nav-Normal-a28a47695855.png)  
   Settings_Nav-Normal comparison
 * Changed: **Folders_TLF-Normal_Public** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479860518327-c70ceccceb9a-a28a47695855-linux/Folders_TLF-Normal_Public-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479860518327-c70ceccceb9a-a28a47695855-linux/Folders_TLF-Normal_Public-a28a47695855.png)  
   Folders_TLF-Normal_Public comparison
 * Changed: **App_Tab_Bar-Search_Active** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479860518327-c70ceccceb9a-a28a47695855-linux/App_Tab_Bar-Search_Active-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479860518327-c70ceccceb9a-a28a47695855-linux/App_Tab_Bar-Search_Active-a28a47695855.png)  
   App_Tab_Bar-Search_Active comparison
 * Changed: **App_Tab_Bar-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479860518327-c70ceccceb9a-a28a47695855-linux/App_Tab_Bar-Normal-c70ceccceb9a.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479860518327-c70ceccceb9a-a28a47695855-linux/App_Tab_Bar-Normal-a28a47695855.png)  
   App_Tab_Bar-Normal comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 0cf34b1b65b1...a28a47695855 introduce visual changes on linux. <details><summary>:mag_right: 1 new, 6 changed</summary>

 * Added: **Profile-Your_Profile_Following_Tab**  
   Profile-Your_Profile_Following_Tab rendered
 * Changed: **Settings_Nav-Normal_Good_Banner** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480359210622-0cf34b1b65b1-a28a47695855-linux/Settings_Nav-Normal_Good_Banner-0cf34b1b65b1.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480359210622-0cf34b1b65b1-a28a47695855-linux/Settings_Nav-Normal_Good_Banner-a28a47695855.png)  
   Settings_Nav-Normal_Good_Banner comparison
 * Changed: **Settings_Nav-Normal_Bad_Banner** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480359210622-0cf34b1b65b1-a28a47695855-linux/Settings_Nav-Normal_Bad_Banner-0cf34b1b65b1.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480359210622-0cf34b1b65b1-a28a47695855-linux/Settings_Nav-Normal_Bad_Banner-a28a47695855.png)  
   Settings_Nav-Normal_Bad_Banner comparison
 * Changed: **Settings_Nav-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480359210622-0cf34b1b65b1-a28a47695855-linux/Settings_Nav-Normal-0cf34b1b65b1.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480359210622-0cf34b1b65b1-a28a47695855-linux/Settings_Nav-Normal-a28a47695855.png)  
   Settings_Nav-Normal comparison
 * Changed: **Folders_TLF-Normal_Public** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480359210622-0cf34b1b65b1-a28a47695855-linux/Folders_TLF-Normal_Public-0cf34b1b65b1.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480359210622-0cf34b1b65b1-a28a47695855-linux/Folders_TLF-Normal_Public-a28a47695855.png)  
   Folders_TLF-Normal_Public comparison
 * Changed: **App_Tab_Bar-Search_Active** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480359210622-0cf34b1b65b1-a28a47695855-linux/App_Tab_Bar-Search_Active-0cf34b1b65b1.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480359210622-0cf34b1b65b1-a28a47695855-linux/App_Tab_Bar-Search_Active-a28a47695855.png)  
   App_Tab_Bar-Search_Active comparison
 * Changed: **App_Tab_Bar-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480359210622-0cf34b1b65b1-a28a47695855-linux/App_Tab_Bar-Normal-0cf34b1b65b1.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480359210622-0cf34b1b65b1-a28a47695855-linux/App_Tab_Bar-Normal-a28a47695855.png)  
   App_Tab_Bar-Normal comparison
</summary>

**Comment by chromakode** [comment]
@keybase/react-hackers bump!

File: shared/route-tree/__tests__/route-tree.test.js [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+    const newRouteState2 = routeSetState(demoRouteDef, newRouteState, ['foo'], {another: 'thing'})
+    expect(newRouteState2).toEqual(new RouteStateNode({
+      selected: 'foo',
+      children: I.Map({
+        foo: new RouteStateNode({selected: null, state: I.Map({state: 'value', another: 'thing'})}),
+      }),
+    }))
+  })
+
+  it('throws when given a path that does\'t exist', () => {
```
Comment (MarcoPolo) [inline shared/route-tree/__tests__/route-tree.test.js]: 🐑  `doesn't`

**Comment by keybase-ci-visdiff** [comment]
The commits 6ed1c577a56c...a2633a5edfbc introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Paper_Key-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480387709228-6ed1c577a56c-a2633a5edfbc-linux/Paper_Key-Normal-6ed1c577a56c.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1480387709228-6ed1c577a56c-a2633a5edfbc-linux/Paper_Key-Normal-a2633a5edfbc.png)  
   Paper_Key-Normal comparison
</summary>
